lib/lint.ml

```diff
@@ -1,17 +1,23 @@
 open Base
-open O
 module Sexp = Sexplib.Sexp
 
+type lint = (unit, [ `Msg of string ]) Result.t
+
+type check = {
+  label : string;
+  f : unit -> [ `Finished of lint | `Recursive of lint * check list ];
```

Comment:
Could you document this type a bit, giving example of when to use each of the variants would be good for instance.

Reply:
I've added some explanation to it. It is not a great type but I wanted it to allow it to fan out to more, recursive checks.

Reply:
Ah yeah thanks! This is crystal clear now!